### PR TITLE
KUI-1256 - remove unused code/dependencies and fix eslint

### PR DESCRIPTION
**Pull request by belanglos**
This PR removes unused code and unused dependencies (or peerDependencies that do not have to explicitly put into `dependencies` anymore). 

It also handles all eslint-errors.

### CODE REVIEW

File: public/js/app/pages/CoursePage.jsx [hunk 1]
```diff
@@ -163,6 +163,8 @@ function CoursePage() {
             </div>
           )}
 
+          <InfoModal title="Test" infoText="<b>yea test</b>" type="html" closeLabel="Close" ariaLabel="SOme aria" />
```
Comment (karlandindrakryggen) [inline public/js/app/pages/CoursePage.jsx:166]: _yea test_ - this wasn't supposed to be included?

File: server/apiCalls/transformers/offerings.js [hunk 2]
```diff
@@ -66,11 +65,14 @@ function _sortOfferedSemesters(offeredSemesters) {
  */
 function _findCourseStartEndDates(courseOfferedSemesters) {
   const offeredSemesters = Array.isArray(courseOfferedSemesters) ? courseOfferedSemesters : []
-  const { length, 0: firstOffering = {}, [length - 1]: lastOffering = {} } = _sortOfferedSemesters(offeredSemesters)
+  const sortedArray = _sortOfferedSemesters(offeredSemesters)
 
-  const { start_date: courseStartDate = '' } = firstOffering
+  const firstOffering = sortedArray.at(0) ?? {}
+  const lastOffering = sortedArray.at(-1) ?? {}
```
Comment (karlandindrakryggen) [inline server/apiCalls/transformers/offerings.js:71]: This is much easier to understand ⭐️

File: server/controllers/__tests__/courseCtrl.test.js [hunk 3]
```diff
@@ -58,6 +58,8 @@ jest.mock('../../apiCalls/koppsCourseData', () => ({ getKoppsCourseData: () => (
 jest.mock('../../apiCalls/ugRestApi', () => ({
   getCourseEmployees: jest.fn(() => ({ examiners: '<p>Examiner 1 </p>' })),
 }))
+// `errorResponse` is clearly used below
```
Comment (karlandindrakryggen) [inline server/controllers/__tests__/courseCtrl.test.js:61]: Is it clearly used? It is assigned, but never read (after you removed ` if (errorResponse) console.error('errorResponse', errorResponse)` below).

**Review by karlandindrakryggen — APPROVED** [review]
Nice! Looks good!

Remove "yea test" before merge, and maybe have a look at `errorResponse´ that is or isn't clearly used. See comments.

File: server/controllers/__tests__/courseCtrl.test.js (hunk 3 above)
Comment (belanglos) [reply server/controllers/__tests__/courseCtrl.test.js]: Fixed c77f3af

File: public/js/app/pages/CoursePage.jsx (hunk 1 above)
Comment (belanglos) [reply public/js/app/pages/CoursePage.jsx]: Fixed 0098888

**Pull request merged**